element-connector/src/test/java/org/eclipse/californium/elements/tcp/TcpCorrelationTest.java

```diff
@@ -0,0 +1,322 @@
+		client.setRawDataReceiver(clientCatcher);
+		server.start();
+		client.start();
+
+		SimpleMessageCallback clientCallback = new SimpleMessageCallback();
+		RawData msg = ConnectorTestUtil.createMessage(server.getAddress(), 100, clientCallback);
+
+		client.send(msg);
+		serverCatcher.blockUntilSize(1);
+
+		CorrelationContext clientContext = clientCallback.getCorrelationContext();
+		assertThat("no TCP Correlation Context", clientContext, is(instanceOf(TcpCorrelationContext.class)));
+		assertThat(clientContext.get(TcpCorrelationContext.KEY_CONNECTION_ID), is(not(isEmptyOrNullString())));
+
+		// Response message must go over the same connection client already
+		// opened
+		SimpleMessageCallback serverCallback = new SimpleMessageCallback();
+		msg = ConnectorTestUtil.createMessage(serverCatcher.getMessage(0).getInetSocketAddress(), 100, serverCallback);
+		server.send(msg);
+		clientCatcher.blockUntilSize(1);
+
+		CorrelationContext serverContext = serverCallback.getCorrelationContext();
+		assertThat("Serverside no TCP Correlation Context", serverContext, is(instanceOf(TcpCorrelationContext.class)));
+		assertThat(serverContext.get(TcpCorrelationContext.KEY_CONNECTION_ID), is(not(isEmptyOrNullString())));
+
+		// check response correlation context
+		CorrelationContext responseContext = clientCatcher.getMessage(0).getCorrelationContext();
+		assertThat("no response TCP Correlation Context", responseContext, is(instanceOf(TcpCorrelationContext.class)));
+		assertThat(responseContext, is(clientContext));
+		assertThat(responseContext.get(TcpCorrelationContext.KEY_CONNECTION_ID),
+				is(clientContext.get(TcpCorrelationContext.KEY_CONNECTION_ID)));
+
+		// send next message
+		clientCallback = new SimpleMessageCallback();
+		msg = ConnectorTestUtil.createMessage(server.getAddress(), 100, clientCallback);
+
+		client.send(msg);
+
+		CorrelationContext context2 = clientCallback.getCorrelationContext(ConnectorTestUtil.CONTEXT_TIMEOUT_IN_MS);
+		assertThat("no TCP Correlation Context", context2, is(instanceOf(TcpCorrelationContext.class)));
+		assertThat(context2, is(clientContext));
+		assertThat(context2.get(TcpCorrelationContext.KEY_CONNECTION_ID),
+				is(clientContext.get(TcpCorrelationContext.KEY_CONNECTION_ID)));
+	}
+
+	@Test
+	public void correlationContextReconnectTimeout() throws Exception {
+		TcpServerConnector server = new TcpServerConnector(new InetSocketAddress(0),
+				ConnectorTestUtil.NUMBER_OF_THREADS, ConnectorTestUtil.IDLE_TIMEOUT_RECONNECT_IN_S);
+		TcpClientConnector client = new TcpClientConnector(ConnectorTestUtil.NUMBER_OF_THREADS,
+				ConnectorTestUtil.CONECTION_TIMEOUT_IN_MS, ConnectorTestUtil.IDLE_TIMEOUT_RECONNECT_IN_S);
+
+		cleanup.add(server);
+		cleanup.add(client);
+
+		Catcher serverCatcher = new Catcher();
+		Catcher clientCatcher = new Catcher();
+		server.setRawDataReceiver(serverCatcher);
+		client.setRawDataReceiver(clientCatcher);
+		server.start();
+		client.start();
```

Comment:
It looks like the code for setting up the client and server is needed several times. Can you re-factor it into a separate method?